### PR TITLE
feat: ORV2-1992 Limit date range to 31 days for reports

### PR DESCRIPTION
**Pull request by krishnan-aot**
# Description
Max Range of 31 days:
image

Error message when user tries a value beyond 31 days:
image

Restricted range when user opens the calendar
image

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] npm run test
- [x] npm run test:e2e

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1230-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1230-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1230-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1230-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://onroutebc-1230-frontend.apps.silver.devops.gov.bc.ca)
- [Vehicles](https://onroutebc-1230-vehicles.apps.silver.devops.gov.bc.ca/api)
- [Dops](https://onroutebc-1230-dops.apps.silver.devops.gov.bc.ca/api)
- [TPS-Migration](https://onroutebc-1230-tps-migration.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/onroutebc/actions/workflows/analysis.yml)

After merge, new images are promoted to:
- [Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/features/idir/reporting/forms/subcomponents/ReportDateTimePickers.tsx [hunk 1]
```diff
@@ -2,17 +2,25 @@ import { FormControl, FormLabel } from "@mui/material";
...
 import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
 import dayjs, { Dayjs } from "dayjs";
+import duration from "dayjs/plugin/duration";
 import { useFormContext } from "react-hook-form";
 
 import { PaymentAndRefundSummaryFormData } from "../../types/types";
 import { RequiredOrNull } from "../../../../../common/types/common";
 
+dayjs.extend(duration);
+const THIRTY_DAYS = 30;
```
Comment (erikataot) [inline frontend/src/features/idir/reporting/forms/subcomponents/ReportDateTimePickers.tsx]: do we need to handle 28 or 31 day months? would the user just be expecting here to select a month limit regardless of day limit?

Comment (praju-aot) [reply frontend/src/features/idir/reporting/forms/subcomponents/ReportDateTimePickers.tsx]: We might have to handle for 31 days so its easier on the user

Comment (krishnan-aot) [reply frontend/src/features/idir/reporting/forms/subcomponents/ReportDateTimePickers.tsx]: Replaced with 31 days as the max limit

**Review by praju-aot — APPROVED** [review]
Approved

**Pull request merged**